bears/rest/RSTcheckBear.py

```diff
@@ -32,16 +33,30 @@ class RSTcheckBear:
 
     @staticmethod
     def create_arguments(filename, file, config_file,
-                         code_block_language_ignore: list=()):
+                         code_block_language_ignore: list=(),
+                         directive_ignore: typed_list(str)=(),
+                         role_ignore: typed_list(str)=()):
         """
         :param code_block_language_ignore:
             Comma seperated list for languages of which code blocks
             should be ignored. Code block of following languages
             can be detected: ``bash``, ``c``, ``cpp``, ``json``,
             ``python``, ``rst``.
+        :param directive_ignore:
+             Comma separated list of directives to be ignored. Can be
+             used to ignore custom directives.
+        :param role_ignore:
+             Comma separated list of roles to be ignored. Can be used
+             to ignore custom roles.
         """
         args = ()
         if code_block_language_ignore:
             args = ('--ignore-language=' +
                     ','.join(code_block_language_ignore),)
-        return args + (filename,)
+        if directive_ignore:
+            directives = ','.join(part.strip() for part in directive_ignore)
```

Comment:
is stripping actually necessary?

Reply:
It is necessary when you pass it to rstcheck, but here we don't know in what way user will pass the arguments

Reply:
maybe coala strips such whitespaces automatically away (also trailing ones behind a comma), we should test that before doing another strip (which would be just waste of execution time and if the user explicitly wants to set whitespaces using escapes - although unlikely here - we would properly fail)

Reply:
Okay I tested it, you can safely leave the `strip()`. Whitespaces inside the lists you pass via a coafile are automatically stripped from each element, so these are the same for example:
```
directive_ignore = a,b      , c     ,     d, e
directive_ignore = a, b, c, d, e
```

Reply:
Okay, but we want is to have "a,b,c,d,e" and not "a, b, c, d, e". `rstcheck` doesn't allow spaces between the names of the directives. Also, I saw stripping done in many bear files.

Reply:
^^ doesn't strip() only remove the whitespaces at the end/beginning of a string? A user would not most likely have an escaped whitespace at the end of a string.

Reply:
yes, however coala already does this for you, you don't need to strip again :+1: As you've specified a `typed_list(str)`